### ISSUE TITLE
First order approximate typo?

### ISSUE THREAD

**Issue by Opdoop** [cnguyen10/few_shot_meta_learning#6]
https://github.com/cnguyen10/few_shot_meta_learning/blob/a38e45cb5dab7ec296709afb1d1120eab87e7907/classification_meta_learning.py#L454

The `inputs` always set to `q_params`, no matter the first-order is true or false. Is this a typo?
```
            if config['first_order']:
                all_grads = torch.autograd.grad(
                    outputs=loss,
                    inputs=q_params,
                    retain_graph=config['train_flag']
                )
            else:
                all_grads = torch.autograd.grad(
                    outputs=loss,
                    inputs=q_params,
                    create_graph=config['train_flag']
                )
```

**Comment by cnguyen10** [comment]
The one mentioned is in the task-adaptation (inner loop) step, where gradient descent is used to optimize the task-specific parameter (denoted as `q` in the code). Hence, we have to calculate the gradient w.r.t. `q`. The first order or higher order is at the argument `retain_graph` or `create_graph`. According to the document of PyTorch, `retain_graph` keeps the graph used to compute the grad, while `create_graph` will construct the graph of the derivative. In the case of `create_graph`, higher order derivative will be calculated when a `backward()` is subsequently called.

I think I know what went wrong here. This is a bug caused by `argparse`, where `boolean` variables are always set to `True` no matter what. Therefore, when you change the `first_order` to `False`, it would still be the same. I corrected it in the new code. Please make a `git pull` to get the latest update. You can give higher order a try by setting the `first_order` as False. Note that this will lead to a higher amount of memory and longer computation time.
